temba/api/serializers.py

```
@@ -53,6 +53,15 @@ def from_native(self, data):
         else:
             raise ValidationError("Invalid, must be array: %s" % data)
 
+class WriteSerializer(serializers.Serializer):
+
+    def restore_fields(self, data, files):
+
+        if data is not None and not isinstance(data, dict):
+            self._errors['non_field_errors'] = ['Invalid data']
```

All of our write serializers expect a posted JSON object right? Then I think `data=None` should throw an error here, i.e just check `if not isinstance(data, dict)`.

Does creating an error here prevent further processing by the serializer?

Maybe also make error message more descriptive like *"Request body should be a single JSON object"*